src/you_get/extractors/netease.py

```diff
@@ -106,18 +107,41 @@ def netease_video_download(vinfo, output_dir='.', info_only=False):
                             output_dir=output_dir, info_only=info_only)
 
 def netease_song_download(song, output_dir='.', info_only=False, playlist_prefix=""):
+    def _download():
+        netease_download_common(title, url_best,
+                                output_dir=output_dir, info_only=info_only)
+
     title = "%s%s. %s" % (playlist_prefix, song['position'], song['name'])
     songNet = 'p' + song['mp3Url'].split('/')[2][1:]
 
-    if 'hMusic' in song and song['hMusic'] != None:
-        url_best = make_url(songNet, song['hMusic']['dfsId'])
-    elif 'mp3Url' in song:
-        url_best = song['mp3Url']
-    elif 'bMusic' in song:
-        url_best = make_url(songNet, song['bMusic']['dfsId'])
+    k = 0
+    while k < 3:
```

Comment:
Use while loop is to try three different url, when one url downloaded fails, try the other two, `url_best` may not be `None`, but will fail to download (HTTP 404). If you have other more concise way, please advise, thanks!